### ISSUE TITLE
Invalid hash given

### ISSUE THREAD

**Issue by jonealex** [Hexxeh/rpi-update#303]
Hello everyone,

Since a few weeks I can't update using `sudo rpi-update`, it throws the error `Invalid hash given`:

![image](https://user-images.githubusercontent.com/6763026/83418171-70b5e180-a423-11ea-82d3-d1fb868b18b0.png)

How can I solve it? Thank you in advance.

Kind regards.

**Comment by pelwell** [comment]
Duplicate of #277.

**Comment by jonealex** [comment]
**UPDATE:**

Changing DNS and using ISP defaults now I can "proceed" but results in the same error:

![image](https://user-images.githubusercontent.com/6763026/83421031-c1c7d480-a427-11ea-8744-4b7cb04b8c75.png)

Note: I'm not using pi-hole.

**Comment by popcornmix** [comment]
Can you report output of:
```
curl -s -L https://api.github.com/repos/Hexxeh/rpi-firmware/commits/master
```
and
```
wget https://github.com/Hexxeh/rpi-firmware/tarball/master
```

**Comment by jonealex** [comment]
> Can you report output of:
> 
> ```
> curl -s -L https://api.github.com/repos/Hexxeh/rpi-firmware/commits/master
> ```
> 
> and
> 
> ```
> wget https://github.com/Hexxeh/rpi-firmware/tarball/master
> ```

**curl**:

```
pi@netbites:~ $ curl -s -L https://api.github.com/repos/Hexxeh/rpi-firmware/commits/master
{
  "sha": "b2aabc3ac7852481cb7833ce068f65eaa4476a2f",
  "node_id": "MDY6Q29tbWl0NDA5NTgxMTpiMmFhYmMzYWM3ODUyNDgxY2I3ODMzY2UwNjhmNjVlYWE0NDc2YTJm",
  "commit": {
    "author": {
      "name": "popcornmix",
      "email": "popcornmix@gmail.com",
      "date": "2020-06-01T12:32:13Z"
    },
    "committer": {
      "name": "popcornmix",
      "email": "popcornmix@gmail.com",
      "date": "2020-06-01T12:32:13Z"
    },
    "message": "firmware: il: isp: Ensure HR output is active and ISP is open before starting a frame\nfirmware: isp_ctrl: Fail in start_[raw|yuv]_frame if ISP is not idle\nSee: https://www.raspberrypi.org/forums/viewtopic.php?f=28&t=275489",
    "tree": {
      "sha": "f72b6e08bdaf4b2028e99567110014517f898623",
      "url": "https://api.github.com/repos/Hexxeh/rpi-firmware/git/trees/f72b6e08bdaf4b2028e99567110014517f898623"
    },
    "url": "https://api.github.com/repos/Hexxeh/rpi-firmware/git/commits/b2aabc3ac7852481cb7833ce068f65eaa4476a2f",
    "comment_count": 0,
    "verification": {
      "verified": false,
      "reason": "unsigned",
      "signature": null,
      "payload": null
    }
  },
  "url": "https://api.github.com/repos/Hexxeh/rpi-firmware/commits/b2aabc3ac7852481cb7833ce068f65eaa4476a2f",
  "html_url": "https://github.com/Hexxeh/rpi-firmware/commit/b2aabc3ac7852481cb7833ce068f65eaa4476a2f",
  "comments_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/commits/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/comments",
  "author": {
    "login": "popcornmix",
    "id": 1053730,
    "node_id": "MDQ6VXNlcjEwNTM3MzA=",
    "avatar_url": "https://avatars3.githubusercontent.com/u/1053730?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/popcornmix",
    "html_url": "https://github.com/popcornmix",
    "followers_url": "https://api.github.com/users/popcornmix/followers",
    "following_url": "https://api.github.com/users/popcornmix/following{/other_user}",
    "gists_url": "https://api.github.com/users/popcornmix/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/popcornmix/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/popcornmix/subscriptions",
    "organizations_url": "https://api.github.com/users/popcornmix/orgs",
    "repos_url": "https://api.github.com/users/popcornmix/repos",
    "events_url": "https://api.github.com/users/popcornmix/events{/privacy}",
    "received_events_url": "https://api.github.com/users/popcornmix/received_events",
    "type": "User",
    "site_admin": false
  },
  "committer": {
    "login": "popcornmix",
    "id": 1053730,
    "node_id": "MDQ6VXNlcjEwNTM3MzA=",
    "avatar_url": "https://avatars3.githubusercontent.com/u/1053730?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/popcornmix",
    "html_url": "https://github.com/popcornmix",
    "followers_url": "https://api.github.com/users/popcornmix/followers",
    "following_url": "https://api.github.com/users/popcornmix/following{/other_user}",
    "gists_url": "https://api.github.com/users/popcornmix/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/popcornmix/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/popcornmix/subscriptions",
    "organizations_url": "https://api.github.com/users/popcornmix/orgs",
    "repos_url": "https://api.github.com/users/popcornmix/repos",
    "events_url": "https://api.github.com/users/popcornmix/events{/privacy}",
    "received_events_url": "https://api.github.com/users/popcornmix/received_events",
    "type": "User",
    "site_admin": false
  },
  "parents": [
    {
      "sha": "85ac48cf41a627c4ebb32fb376d6a13c21271479",
      "url": "https://api.github.com/repos/Hexxeh/rpi-firmware/commits/85ac48cf41a627c4ebb32fb376d6a13c21271479",
      "html_url": "https://github.com/Hexxeh/rpi-firmware/commit/85ac48cf41a627c4ebb32fb376d6a13c21271479"
    }
  ],
  "stats": {
    "total": 0,
    "additions": 0,
    "deletions": 0
  },
  "files": [
    {
      "sha": "88363225f96f737abfd154b411ee2df02e04d61d",
      "filename": "fixup.dat",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup.dat",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup.dat",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/fixup.dat?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "0cbef6575d8428b0075397c4121bbf0a938df0b4",
      "filename": "fixup4.dat",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup4.dat",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup4.dat",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/fixup4.dat?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "c172df42be3a842413e045d03cd45f7211f36f48",
      "filename": "fixup4db.dat",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup4db.dat",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup4db.dat",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/fixup4db.dat?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "32b03c869b18e6bacb4a667117714f8a96e142ed",
      "filename": "fixup4x.dat",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup4x.dat",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup4x.dat",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/fixup4x.dat?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "c911b4bdd674de7dd8e609b8f4d45c352dde3624",
      "filename": "fixup_db.dat",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup_db.dat",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup_db.dat",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/fixup_db.dat?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "7a3ade15dc2dc71ee81f504ac6a7080f310d106f",
      "filename": "fixup_x.dat",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup_x.dat",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/fixup_x.dat",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/fixup_x.dat?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "d40bc010f2c4b8097cf740ae8b30759752b8d578",
      "filename": "start.elf",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start.elf",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start.elf",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/start.elf?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "23dad9cb7e2530d23fea4b0b3a8c2e5219a22aa3",
      "filename": "start4.elf",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start4.elf",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start4.elf",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/start4.elf?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "38b496873fad52e983b523f5a509d0deda626455",
      "filename": "start4cd.elf",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start4cd.elf",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start4cd.elf",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/start4cd.elf?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "1925c424341c77edeed62098f35bbdfaa117bbdf",
      "filename": "start4db.elf",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start4db.elf",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start4db.elf",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/start4db.elf?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "cdd1e36c54733aaedfbc5a1a4a41e8683375a52b",
      "filename": "start4x.elf",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start4x.elf",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start4x.elf",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/start4x.elf?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "13a79fa187104e3eeb94f67c07e14f3988133a5a",
      "filename": "start_cd.elf",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start_cd.elf",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start_cd.elf",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/start_cd.elf?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "4ba20682ecb5b8d442e2deffa344f3fbaff1f621",
      "filename": "start_db.elf",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start_db.elf",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start_db.elf",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/start_db.elf?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    },
    {
      "sha": "2bb7770d367911d611aaadbc43cb0c00982c9a3d",
      "filename": "start_x.elf",
      "status": "modified",
      "additions": 0,
      "deletions": 0,
      "changes": 0,
      "blob_url": "https://github.com/Hexxeh/rpi-firmware/blob/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start_x.elf",
      "raw_url": "https://github.com/Hexxeh/rpi-firmware/raw/b2aabc3ac7852481cb7833ce068f65eaa4476a2f/start_x.elf",
      "contents_url": "https://api.github.com/repos/Hexxeh/rpi-firmware/contents/start_x.elf?ref=b2aabc3ac7852481cb7833ce068f65eaa4476a2f"
    }
  ]
}
```

**wget**:

```
pi@netbites:~ $ wget https://github.com/Hexxeh/rpi-firmware/tarball/master
--2020-06-01 17:36:16--  https://github.com/Hexxeh/rpi-firmware/tarball/master
Resolviendo github.com (github.com)... falló: Nombre o servicio desconocido.
wget: no se pudo resolver la dirección del equipo “github.com”
```

**Comment by popcornmix** [comment]
You've got a problem with your DNS lookups. It can't locate github.com (where we are now).
Either a bug with DNS provider, an invalid setting in your local lookup (e.g. /etc/hosts) or some form of adblocking is behaving badly. It's not something we can fix from here.

To narrow it down try the same wget command from another PC on your local network. Does it fail there too?

**Comment by jonealex** [comment]
> You've got a problem with your DNS lookups. It can't locate github.com (where we are now).
> Either a bug with DNS provider, an invalid setting in your local lookup (e.g. /etc/hosts) or some form of adblocking is behaving badly. It's not something we can fix from here.
> 
> To narrow it down try the same wget command from another PC on your local network. Does it fail there too?

Stranger things... on my main computer (Windows 10) I can resolve it without problems:

```
C:\Users\aleja>ping github.com

Haciendo ping a github.com [140.82.118.3] con 32 bytes de datos:
Respuesta desde 140.82.118.3: bytes=32 tiempo=49ms TTL=51
Respuesta desde 140.82.118.3: bytes=32 tiempo=49ms TTL=51
Respuesta desde 140.82.118.3: bytes=32 tiempo=54ms TTL=51
Respuesta desde 140.82.118.3: bytes=32 tiempo=49ms TTL=51

Estadísticas de ping para 140.82.118.3:
    Paquetes: enviados = 4, recibidos = 4, perdidos = 0
    (0% perdidos),
Tiempos aproximados de ida y vuelta en milisegundos:
    Mínimo = 49ms, Máximo = 54ms, Media = 50ms
```

But on Linux Subsystem (same computer, same local network):

```
jonealex@DESKTOP-DE9UPRV:~$ wget https://github.com/Hexxeh/rpi-firmware/tarball/master
--2020-06-01 17:44:45--  https://github.com/Hexxeh/rpi-firmware/tarball/master
Resolving github.com (github.com)... failed: Name or service not known.
wget: unable to resolve host address ‘github.com’
```

**Comment by popcornmix** [comment]
On linux you can add the line:
```140.82.118.3 github.com```
to `/etc/hosts` which may work around the issue. But it's probably a problem with your router, or something outside of the Pi.

**Comment by jonealex** [comment]
> On linux you can add the line:
> `140.82.118.3 github.com`
> to `/etc/hosts` which may work around the issue. But it's probably a problem with your router, or something outside of the Pi.

Yup, I solved by adding a simple line to `/etc/hosts` but I should talk about with my ISP.

Thank you !
